src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -60,13 +66,23 @@ class Attribute {
                       });
     }
 
-    void set_callback(callback_t callback) {
-        m_callback = std::move(callback);
-    }
-
     // Apply callback to materialize RIC inside graph
-    void operator()(Input<Node> input) const {
-        m_callback(input, *this);
+    void materialize(Input<Node> input) const {
+        if (get_axis() >= input.get_partial_shape().size())
+            throw std::runtime_error("Axis not in range");
+        const auto& axis_dim = input.get_partial_shape()[get_axis()];
+        if (axis_dim.is_dynamic())
+            throw std::runtime_error("Axis dimension is dynamic");
+        auto output = input.get_source_output();
+        // Handle case when the RIC order is default
+        auto order = get_order();
+        if (order.empty()) {
+            order.resize(axis_dim.get_length());
+            std::iota(order.rbegin(), order.rend(), 0);
+        }
+        auto gather = std::make_shared<opset8::Gather>(output, create_const(order), create_const({get_axis()}));
+        input.replace_source_output(gather);
+        // TODO: copy runtime info from RIC sub-graph
```

Comment:
is it ok?

Reply:
It is a missing part even before extending. I've prepared a separate ticket for support it (88597) as sth not very urgent and not trivial to do.